### PR TITLE
Fix OCaml syntax

### LINKED ISSUE #5113: Problem highlighting OCaml code
We found some issues when highlighting OCaml code, especially with:

 * Some number literals (hex, binary, exponential floats)
 * Bracket string literals
 * Some reserved words (sig, module)
 
You can see a working example with current CodeMirror version at http://jsbin.com/yutewuwafa/edit?html,js,output

### PR DESCRIPTION
**Pull request by cprieto**
There were some issues with OCaml syntax:
 * First element in a list is not highlighted (when it is a number)
 * Binary number literal was not highlighted
 * Hexadecimal number literal was not highlighted
 * `module` and `sig` were not highlighted

This closes #5113 

I will suggest, if the OCaml syntax continue growing we should split it in a separate file to avoid issues with F# and other ML languages (for example, standard ML).

### CODE REVIEW

File: mode/mllike/mllike.js [hunk 1]
```diff
@@ -74,9 +81,18 @@ CodeMirror.defineMode('mllike', function(_config, parserConfig) {
       return 'comment';
     }
     if (/\d/.test(ch)) {
-      stream.eatWhile(/[\d]/);
-      if (stream.eat('.')) {
+      if (ch === '0' && stream.eat(/[bB]/)) {
+        stream.eatWhile(/[01]/);
+      } if (ch === '0' && stream.eat(/[xX]/)) {
+        stream.eatWhile(/[0-9a-dA-D]/)
```
Comment (adrianheine) [inline mode/mllike/mllike.js]: What about e and f?

Comment (cprieto) [reply mode/mllike/mllike.js]: I have no idea what you meant.

if it is about the case with `1e-10` that is already covered in line 53 with the patch. Now, with the `f` case I have no idea what is that, OCaml doesn't have a 1f notation (that I know).

Please, if I am misunderstanding what you are pointing out I will appreciate it.

Now that you point that out, I notice I forgot Octal Integer Literals (`0o1234`), I am adding them now.

Comment (adrianheine) [reply mode/mllike/mllike.js]: The line I commented on is parsing hexadecimal literals, which use digits from 0 to f, right?

Comment (cprieto) [reply mode/mllike/mllike.js]: Oh thanks mate!

I really missed that one! thanks man!

Thanks to you I got as well the long integer literals and octal 😄 

Thanks again 👍

**Comment by marijnh** [comment]
Great. I've merged this as 32f63fc3, minus the external theme (most of the demo pages work offline, and might as well continue to do so)

**Pull request closed without merging**